server/models/Collective.ts

```diff
@@ -2383,6 +2385,18 @@ class Collective extends Model<
       await Promise.all(children.map(e => e.changeHost(null, remoteUser, { isChildren: true })));
     }
 
+    // Unlink accounting categories from unpaid expenses, as these categories are host-specific
+    await models.Expense.update(
+      { AccountingCategoryId: null },
+      {
+        where: {
+          CollectiveId: [this.id, ...children.map(c => c.id)],
+          status: { [Op.not]: 'PAID' },
```

Comment:
I'm not sure if we have another validation to prevent a collective change while an Expense is still processing, but theoretically:
```suggestion
          status: { [Op.notIn]: ['PAID', 'PROCESSING'] },
```

Reply:
My thinking for limiting this to `PAID` was:
- If an expense is still processing, we should probably not let the unhost happen
- If it's really processing, then it will be paid by the current host, thus it makes sense to keep the accounting category